pilot/pkg/xds/filters/filters.go

```diff
@@ -229,6 +229,50 @@ var (
 		},
 	}
 
+	SidecarInboundMetadataFilter = &hcm.HttpFilter{
+		Name: MxFilterName,
+		ConfigType: &hcm.HttpFilter_TypedConfig{
+			TypedConfig: protoconv.TypedStructWithFields("type.googleapis.com/io.istio.http.peer_metadata.Config",
+				map[string]any{
+					"downstream_discovery": []any{
+						map[string]any{
+							"istio_headers": map[string]any{},
+						},
+						map[string]any{
+							"workload_discovery": map[string]any{},
```

Comment:
You mentioned its off by default. What makes it off?

Reply:
Lack of bootstrap extension "workload_discovery". Without it, this setting is no-op.

Reply:
Kuat, What do you mean by Lack of bootstrap extension "workload_discovery". Without it, this setting is no-op.? I assume that comment is for workload discovery, not for the metadata exchange?

Reply:
Chatted with Kuat offline,without workload discovery, the filter will still perform HTTP MX header.